webapp/platform/client/src/client4.ts

```diff
@@ -4186,6 +4197,17 @@ export default class Client4 {
             {method: 'delete', body: JSON.stringify(deletionRequest)},
         );
     }
+
+    // This allows the server to know that a given HTTP request occurred during page load or reconnect.
+    // The server then uses this information to store metrics fields based on the request context.
+    // The setTimeout approach is a "best effort" approach that will produce false positives.
+    // A more accurate approach will result in more obtrusive code, which would add risk and maintenance cost.
+    temporarilySetPageLoadContext = (pageLoadContext: PageLoadContext | null) => {
+        this.pageLoadContext = pageLoadContext;
+        setTimeout(() => {
+            this.pageLoadContext = null;
+        }, 5000);
+    }
```

Comment:
This is definitely best than what we have (which is nothing), but it still makes me shiver a bit :joy: I cannot wrap my head around how many requests this will affect. I guess the 5s timeout is just a random number, right? There could be cases where it takes more than 5 seconds to load everything, and maybe those cases are even more important to analyze?
You say that "A more accurate approach will result in more obtrusive code, which would add risk and maintenance cost." Have you had a chance to try how that would look like? I don't see a much better solution either (other than passing `null|reconnnect|page_load` to each and every client function), but maybe there is one :thinking:

Reply:
The way the browser's performance telemetry works, you use `Performance.mark` at the start and end of a performance event and then you use a method like `Performance.measure(start, end)` to measure how much time passed between those. We already have marks in place that we use for the `page_load` time collected and tracked on Looker, so perhaps we could look at where those are done and add/remove a header to all `Client4` requests at those times? It may not catch everything since I'm always a bit skeptical of if that's an accurate view of page load time, but it would at least make it consistent with the existing metrics.

That said, we have also tried to count the number of requests made during that window using `Performance`, but I feel like those numbers were always very low compared to what we actually expected to see